internal/core/model.rs

```diff
@@ -767,16 +767,23 @@ pub struct RepeaterTracker<T: RepeatedItemTree> {
 
 impl<T: RepeatedItemTree> ModelChangeListener for RepeaterTracker<T> {
     /// Notify the peers that a specific row was changed
-    fn row_changed(&self, row: usize) {
-        self.is_dirty.set(true);
+    fn row_changed(self: Pin<&Self>, row: usize) {
         let mut inner = self.inner.borrow_mut();
         let inner = &mut *inner;
         if let Some(c) = inner.instances.get_mut(row.wrapping_sub(inner.offset)) {
-            c.0 = RepeatedInstanceState::Dirty;
+            if !self.model.is_dirty() {
+                if let Some(comp) = c.1.as_ref() {
+                    let model = self.project_ref().model.get_untracked();
```

Comment:
i need self to be Pin here to access the model, Hence the addition of Pin everywhere. (although we could do without because we know the model is not dirty)